src/wmtk/simplex/link_condition.cpp

```diff
@@ -0,0 +1,90 @@
+#include "link_condition.hpp"
+#include "link.hpp"
+#include "open_star.hpp"
+
+namespace wmtk::simplex {
+bool link_condition_closed_trimesh(const TriMesh& mesh, const Tuple& edge)
+{
+    SimplexCollection lnk_a = link(mesh, Simplex(PrimitiveType::Vertex, edge), true); // link(a)
+    SimplexCollection lnk_b = link(
+        mesh,
+        Simplex(PrimitiveType::Vertex, mesh.switch_tuple(edge, PrimitiveType::Vertex)),
+        true); // link(b)
+    SimplexCollection lnk_ab = link(mesh, Simplex(PrimitiveType::Edge, edge), true); // link(ab)
+
+    SimplexCollection lnk_a_lnk_b_intersection = SimplexCollection::get_intersection(lnk_a, lnk_b);
+
+    return SimplexCollection::are_simplex_collections_equal(lnk_a_lnk_b_intersection, lnk_ab);
+}
+
+bool link_condition(const EdgeMesh& mesh, const Tuple& edge)
+{
+    const Tuple edge_switch_v = mesh.switch_tuple(edge, PrimitiveType::Vertex);
+    if (mesh.is_boundary_vertex(edge) && mesh.is_boundary_vertex(edge_switch_v)) {
+        return false;
+    }
+    if (mesh.simplices_are_equal(
+            Simplex(PrimitiveType::Vertex, edge),
+            Simplex(PrimitiveType::Vertex, edge_switch_v))) {
+        return false;
+    }
+    return true;
+}
+
+bool link_condition(const TriMesh& mesh, const Tuple& edge)
+{
+    // step1 check link condition for closed case
+    if (!link_condition_closed_trimesh(mesh, edge)) {
+        return false;
+    }
+
+    // check if dummy vertex w is included in the lhs
+    auto get_boundary_edges = [&mesh](const Tuple& _v) {
+        Simplex input_v(PrimitiveType::Vertex, _v);
+        std::vector<Tuple> ret;
+        // get one_ring_edges from open_star
+        auto one_ring_edges = open_star(mesh, input_v).simplex_vector(PrimitiveType::Edge);
```

Comment:
These are not the one-ring edges by the way. These are the incident edges. One-ring edges are the edges in the link.